gst/nnstreamer/tensor_sparse/tensor_sparse_util.c

```diff
@@ -0,0 +1,207 @@
+/* SPDX-License-Identifier: LGPL-2.1-only */
+/**
+ * GStreamer / NNStreamer Sparse Tensor support
+ * Copyright (C) 2021 Yongjoo Ahn <yongjoo1.ahn@samsung.com>
+ */
+/**
+ * @file	tensor_sparse_util.c
+ * @date	27 Jul 2021
+ * @brief	Util functions for tensor_sparse encoder and decoder.
+ * @see		https://github.com/nnstreamer/nnstreamer
+ * @author	Yongjoo Ahn <yongjoo1.ahn@samsung.com>
+ * @bug		No known bugs except for NYI items
+ */
+
+#ifndef __GST_TENSOR_SPARSE_ENC_H__
+#define __GST_TENSOR_SPARSE_ENC_H__
+
+#include <tensor_common.h>
+#include <tensor_data.h>
+#include "tensor_sparse_util.h"
+
+/**
+ * @brief Make dense tensor with input sparse tensor.
+ * @param[in,out] meta tensor meta structure to be updated
+ * @param[in] in pointer of input sparse tensor data
+ * @return  pointer of Gstmemory with dense tensor data or NULL on error. Caller should handle this newly allocated memory.
+ */
+extern GstMemory *
+gst_tensor_sparse_to_dense (GstTensorMetaInfo * meta, gpointer in)
+{
+  int i;
+  guint8 *output, *input;
+  guint *indices;
+  guint nnz;
+  gsize output_size, element_size = gst_tensor_get_element_size (meta->type);
+  gulong element_count = gst_tensor_get_element_count (meta->dimension);
+
+  output_size = element_size * element_count;
```

Comment:
Using meta, recommend to use gst_tensor_meta_info_get_data_size().
`output_size = gst_tensor_meta_info_get_data_size (meta);`


NNS_TENSOR_RANK_LIMIT is 4, but NNS_TENSOR_META_RANK_LIMIT is 16.

Reply:
> NNS_TENSOR_RANK_LIMIT is 4, but NNS_TENSOR_META_RANK_LIMIT is 16.

Thanks for this info. I changed this part to use `gst_tensor_meta_info_get_data_size`